src/main/java/seedu/address/logic/commands/StartCommand.java

```diff
@@ -75,8 +75,7 @@ public boolean equals(Object other) {
             return new LinkedList<>(model.getFlashCardList());
         }
         CategoryContainsAnyKeywordsPredicate predicate = getSearchTermPredicate();
-        model.updateFilteredFlashCardList(predicate);
-        return new LinkedList<>(model.getFilteredFlashCardList());
+        return new LinkedList<>(model.getFilteredFlashCardListNoCommit(predicate));
```

Comment:
I think that `getFilteredFlashCardListNoCommit(predicate)` is returning a copy of the list. So maybe don't need `new LinkedList<>`

Reply:
I initially considered LinkedList because of the O(1) complexity in removing head of the list, while at the same time improving readability, i.e. in the `FlashCardTestModel`'s `remove(0);` method instead of `remove(list.size() - 1)`.

But now I think it wouldn't be realistic for users to go through 1000s of test FlashCards in 1 sitting so O(n) complexity is acceptable.

Good catch!